markup/goldmark/convert.go

```diff
@@ -194,6 +195,30 @@ func newMarkdown(pcfg converter.ProviderConfig) goldmark.Markdown {
 		))
 	}
 
+	if cfg.Extensions.Extras.Insert.Enable {
```

Comment:
The `extras` has the same structure as Hugo's, so I would just do

```go
extensions = append(extensions, extras.New(
			extras.Config{
                            Mark: extras.MarkConfig{Enable: cfg.Extensions.Extras.Mark.Enable},
                            ...
},
))
```

If all are disabled, you'd end up with a `noop` extension.

Reply:
OK, I see I was a little unclear about this. The current setup allows for setting up 1 extension in one go without any conditionals:

```go
extensions = append(extensions, extras.New(
	extras.Config{
	   Insert: extras.InsertConfig{Enable: cfg.Extensions.Extras.Insert.Enable},
           Mark: extras.MarConfig{Enable: cfg.Extensions.Extras.Mark.Enable},
           // ... the rest of the extras.
	},
))
```

The above is, in my head,

1. Easier to reason about.
2. It would allow potential upstream improvements, maybe the implementation could be sped up by making it into one parser instance?

Reply:
Gotcha.